core/src/banking_stage/transaction_scheduler/transaction_packet_container.rs

```diff
@@ -0,0 +1,324 @@
+use {
+    super::transaction_priority_id::TransactionPriorityId,
+    crate::banking_stage::{
+        immutable_deserialized_packet::ImmutableDeserializedPacket,
+        scheduler_messages::TransactionId, unprocessed_packet_batches::DeserializedPacket,
+    },
+    min_max_heap::MinMaxHeap,
+    solana_sdk::{slot_history::Slot, transaction::SanitizedTransaction},
+    std::collections::HashMap,
+};
+
+/// Simple wrapper to tie a `SanitizedTransaction` with the maximum slot
+/// it can be processed at without re-sanitization.
+pub(crate) struct SanitizedTransactionTTL {
+    pub(crate) transaction: SanitizedTransaction,
+    pub(crate) max_age_slot: Slot,
+}
+
+pub(crate) struct TransactionPacketContainer {
+    /// Unscheduled unique identifiers ordered by priority
+    priority_queue: MinMaxHeap<TransactionPriorityId>,
+    /// Map from `TransactionId` to `SanitizedTransactionTTL`.
+    /// Contains transactions that have not yet been scheduled for consuming.
+    id_to_transaction_ttl: HashMap<TransactionId, SanitizedTransactionTTL>,
+    /// Map from `TransactionId` to `DeserializedPacket`.
+    /// Contains packets for the entire lifetime of packet/transaction.
+    /// These will only be removed once a transaction has finished processing,
+    /// or if the packets are forwarded without holding.
+    id_to_packet: HashMap<TransactionId, DeserializedPacket>,
+}
+
+impl TransactionPacketContainer {
+    pub(crate) fn with_capacity(capacity: usize) -> Self {
+        Self {
+            priority_queue: MinMaxHeap::with_capacity(capacity),
+            id_to_transaction_ttl: HashMap::with_capacity(capacity),
+            id_to_packet: HashMap::with_capacity(capacity),
```

Comment:
Mmm, sync 3 inner containers feels tricky. Would add rules and enforcing them necessary?